modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/getFields.ts

```diff
@@ -120,14 +134,25 @@ function getValidFields({
 				!contextPath.includes(FDS_ARRAY_FIELD_NAME_DELIMITER);
 
 			if (targetSchemaName && !visitedFields.includes(targetSchemaName)) {
+				const fullPath = parentPath
+					? `${parentPath}/${propertyKey}`
+					: propertyKey;
+
 				field.children = getValidFields({
 					contextPath: field.name,
+					parentPath: fullPath,
 					schemaName: targetSchemaName,
 					schemas,
 					visitedFields: [...visitedFields, targetSchemaName],
+					xFilterable,
 				});
 			}
 
+			const fullPath = parentPath
+				? `${parentPath}/${propertyKey}`
+				: propertyKey;
+			field.filterable = xFilterable.includes(fullPath);
```

Comment:
This is probably fine for our internal endpoints, I assume all of them will now have this explicit flag. But, if customer made their own rest application, it is possible that they will suddenly see no fields. Even if they update rest application, they may have active filters in a data set that is no longer valid. I _think_ that is not breaking, they just cannot update it in DSM. @dsanz @herrardo Am I understanding this correctly? If so, is this risk acceptable? (I think it is, but it's a product question)

Reply:
This risk can happen on already created filters on pre-existing datasets: after this change in DSM, some filters could not be created from scratch. But if it existed previously, we shall allow to update it like any other.

Another case to consider is how to deal with the field selection modal outside filters. I specifically refer to the field selection when mapping fields to table/card/list. There, field availability does not depend on filterability.